lib/dependency-manager-adapters/npm.js

```diff
@@ -54,10 +73,19 @@ module.exports = CoreObject.extend({
   },
   cleanup: function() {
     var adapter = this;
-    return adapter._restoreOriginalDependencies().then(function() {
+    var backupYarnLockPresent = fs.existsSync(path.join(this.cwd, this.yarnLockBackupFileName));
+    return this._restoreOriginalDependencies(backupYarnLockPresent).then(function() {
       debug('Remove backup package.json and node_modules');
-      return RSVP.all([rimraf(path.join(adapter.cwd, adapter.packageJSONBackupFileName)),
-        rimraf(path.join(adapter.cwd, adapter.nodeModulesBackupLocation))]);
+
+      var cleanupTasks = [rimraf(path.join(adapter.cwd, adapter.packageJSONBackupFileName)),
+        rimraf(path.join(adapter.cwd, adapter.nodeModulesBackupLocation))];
+
+      if (backupYarnLockPresent) {
+        cleanupTasks.push(rimraf(path.join(adapter.cwd, adapter.yarnLockBackupFileName)));
+        adapter.noLock = false;
```

Comment:
I still think `noLock` being passed as a parameter to `_install` would make things clearer than a flag.